### DISCUSSION TITLE
How to decorate methods with kopf decorators (e.g. create)?

### DISCUSSION THREAD

**Discussion by pepedocs** [nolar/kopf#938] [Q&A] [answered]
Hi,

How do I decorate class functions or methods with kopf decorators (e.g. create)?

For example;

```
    @kopf.on.create("Cloud")
    def on_create_cloud(self, spec, name, namespace, **kwargs):
```

I get this error;

```
    result = self.fn(*self.args, **self.kwargs)
TypeError: on_create_cloud() missing 1 required positional argument: 'self'

```

**Comment by mledoze** [comment, marked as answer]
Hi, 

Kopf decorators are not meant to be used on class methods, see here https://github.com/nolar/kopf/issues/849#issuecomment-959846403 for an explanation.

**Reply by pepedocs** [reply]
Thanks for the reply that is very useful.

Although I think a singleton class's case is IMO valid. It will only create one instance but I also don't want a static method. Using a class simplifies the namespace of the operator-related funcs or variables.

In any case I found a simple way by just wrapping the handler and calling the operator's methods.
